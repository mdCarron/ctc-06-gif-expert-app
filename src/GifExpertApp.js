import React, { Fragment, useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GifExpertApp = ({ defaultCategorias = [] }) => {
  // const [categorias, setCategorias] = useState(["Metroid"]);
  const [categorias, setCategorias] = useState(defaultCategorias);

  return (
    <Fragment>
      <h2>GifExpertApp</h2>
      <AddCategory setCategorias={setCategorias} categorias={categorias} />
      <hr />
      <ul>
        {categorias.map((categoria) => (
          <GifGrid key={categoria} categoria={categoria} />
        ))}
      </ul>
    </Fragment>
  );
};

export default GifExpertApp;
