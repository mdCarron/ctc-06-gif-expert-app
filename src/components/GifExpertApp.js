import React, { Fragment } from "react";
import PropTypes from "prop-types";

const GifExpertApp = () => {
  return (
    <Fragment>
      <h2>GifExpertApp</h2>
      <hr />
    </Fragment>
  );
};

GifExpertApp.propTypes = {};

export default GifExpertApp;
