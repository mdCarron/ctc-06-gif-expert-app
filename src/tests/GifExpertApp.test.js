import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("testing en componente <GifExpertApp />", () => {
  test("should hacer match con snapshot", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should iniciar con las categorias indicadas", () => {
    const categorias = ["super nintendo", "nes"];
    const wrapper = shallow(<GifExpertApp defaultCategorias={categorias} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categorias.length);
  });
});
