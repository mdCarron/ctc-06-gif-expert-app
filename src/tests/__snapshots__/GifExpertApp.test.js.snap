// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing en componente <GifExpertApp /> should hacer match con snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    categorias={Array []}
    setCategorias={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`testing en componente <GifExpertApp /> should iniciar con las categorias indicadas 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    categorias={
      Array [
        "super nintendo",
        "nes",
      ]
    }
    setCategorias={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      categoria="super nintendo"
      key="super nintendo"
    />
    <GifGrid
      categoria="nes"
      key="nes"
    />
  </ul>
</Fragment>
`;
